scripts/segmentation_HALO-20240822a.md

```diff
@@ -0,0 +1,389 @@
+
+pace_track = _pace_track.where(
+        (_pace_track.lat > ds.lat.min()-2) & (_pace_track.lat < ds.lat.max()+2) &
+        (_pace_track.lon > ds.lon.min()-2) & (_pace_track.lon < ds.lon.max()+2),
+        drop=True)
+```
+
+### Get METEOR track
+select maybe only the track from the respective flight day
+
+```python
+from orcestra.meteor import get_meteor_track
+
+meteor_track = get_meteor_track().sel(time=slice(takeoff, landing))
+```
+
+## Overview plot: HALO track, EC meeting point, and dropsonde locations
+
+```python
+plt.plot(ds.lon.sel(time=slice(takeoff, landing)), ds.lat.sel(time=slice(takeoff, landing)), label="HALO track")
+plt.scatter(ds_drops.lon, ds_drops.lat, s=10, c="k", label="dropsondes")
+plt.plot(ec_track.lon, ec_track.lat, c='C1', ls='dotted')
+plt.plot(ds.lon.sel(time=t_ec, method="nearest"), ds.lat.sel(time=t_ec, method="nearest"), marker="*", ls=":", label="EC meeting point")
+#plt.plot(pace_track.lon, pace_track.lat, c="C2", ls=":", label="PACE track")
+plt.plot(meteor_track.lon, meteor_track.lat, c="C4", ls="-.", label="METEOR track")
+plt.xlabel("longitude / °")
+plt.ylabel("latitude / °")
+plt.legend();
+```
+
+## Interactive plots
+
+```python
+ds["alt"].hvplot()
+```
+
+```python
+ds["roll"].hvplot()
+```
+
+```python
+ds["heading"].hvplot()
+```
+
+## Segments
+
+defined as a tuple of time slice (`start`, `end`) , segment `kind`, `name`, `remarks`.
+
+* in case of irregularities within a circle, 1 sec before the first and after the last sonde are chosen as start and end times
+* use the list of `remarks` to state any deviations, also with respective times
+
+Alternatively, you can also define the segments as dictionaries which also allows to add further attributes to single segments, e.g. a `radius` to a `circle` segment. At the end of the following code block all segments will be normalized by the `parse_segments` function.
+
+```python
+seg1 = (
+    slice("2024-08-22T11:26:22", "2024-08-22T11:29:11"),
+    ["straight_leg", "ascent"],
+    "ascending ferry",
+)
+
+seg2 = (
+    slice("2024-08-22T11:31:15", "2024-08-22T11:52:44"),
+    ["straight_leg", "ascent", "ec_track"],
+    "southward ascent on EC track",
+)
+
+seg3 = (
+    slice("2024-08-22T11:52:44", "2024-08-22T12:47:43"),
+    ["straight_leg", "ec_track"],
+    "EC underpass leg"
+)
+
+seg4 = (
+    slice("2024-08-22T12:50:11", "2024-08-22T13:46:03"),
+    ["circle"],
+    "southern counterclockwise circle",
+    ["potential Meteor overpass"]
+)
+
+seg5 = (
+    slice("2024-08-22T13:49:45", "2024-08-22T13:53:56"),
+    ["straight_leg", "ascent", "ec_track"],
+    "northward ascent on EC track",
+)
+
+seg6 = (
+    slice("2024-08-22T13:57:24", "2024-08-22T14:54:18"),
+    ["circle"],
+    "middle counterclockwise circle",
+    ["irregularities: turbulence between 2024-08-22 14:19:07 and 2024-08-22 14:42:09"]
+)
+
+seg7 = (
+    slice("2024-08-22T14:57:17", "2024-08-22T16:04:31"),
+    ["straight_leg"],
+    "northward leg on EC track",
+)
```

Comment:
["irregularity: turbulence between 15:28:10 and 15:28:38 with +3.0 deg roll angle deviation",
    "includes EC meeting point"]